testsuite/tests/win-unicode/mltest.ml

```diff
@@ -0,0 +1,233 @@
+    let rec loop () =
+      match input_line ic with
+      | None ->
+          flog "======== EOF %s ========" file
+      | Some line ->
+          flog "%s" line;
+          loop ()
+    in
+    loop ()
+  in
+  flog "======== %s ========" file;
+  with_open_in file f
+
+let sort_entries = List.sort compare
+
+let get_entries_unix () =
+  let h = Unix.opendir "." in
+  let rec loop acc =
+    match Unix.readdir h with
+    | s -> if s <> Filename.parent_dir_name && s <> Filename.current_dir_name then loop (s :: acc) else loop acc
+    | exception End_of_file -> Unix.closedir h; sort_entries acc
+  in
+  loop []
+
+let get_entries () =
+  let entries = Sys.readdir "." in
+  let entries = Array.to_list entries in
+  let entries = sort_entries entries in
+  assert (entries = get_entries_unix ());
+  entries
+
+let test_stat stat stat_name file =
+  flog "stat with %s on %s" stat_name file;
+  let stats = stat file in
+  assert (stats.Unix.st_kind = Unix.S_REG)
+
+let test_large_file_stat stat stat_name file =
+  flog "%s on \"%s\"" stat_name file;
+  let stats = stat file in
+  assert (stats.Unix.LargeFile.st_kind = Unix.S_REG)
+
+let test_file_exists file =
+  flog "Sys.file_exists on \"%s\"" file;
+  assert (Sys.file_exists file)
+
+let test_rename f name file file_renamed =
+  flog "%s on \"%s\" -> \"%s\"" name file file_renamed;
+  f file file_renamed
+
+let test_call f name file =
+  flog "%s on \"%s\"" name file;
+  f file
+
+let create_and_delete file =
+  let file = file ^ ".txt" in
+  flog "creating and deleting \"%s\"" file;
+  let output_hi oc =
+    output_string oc "hi\n";
+    flush oc
+  in
+  let old_entries = get_entries () in
+  with_open_out file output_hi;
+  let current_entries = get_entries () in
+  assert (current_entries = sort_entries (file :: old_entries));
+  test_stat Unix.stat "Unix.stat" file;
+  test_stat Unix.lstat "Unix.lstat" file;
+  test_large_file_stat Unix.LargeFile.stat "Unix.LargeFile.stat" file;
+  test_large_file_stat Unix.LargeFile.lstat "Unix.LargeFile.lstat" file;
+  test_file_exists file;
+  let rename s = s ^ "-1" in
+  test_rename Unix.rename "Unix.rename" file (rename file);
+  test_file_exists (rename file);
+  test_rename Sys.rename "Sys.rename" (rename file) file;
+  test_file_exists file;
+  test_call Sys.remove "Sys.remove" file;
+  let new_entries = get_entries () in
+  assert (old_entries = new_entries)
+
+let extension file = String.lowercase_ascii (Filename.extension file)
+
+let to_create_and_delete_files =
+  [
+    "верблюды";
+    "骆驼";
+    "\215\167\215\162\215\158\215\156";
+    "\216\167\217\136\217\134\217\185";
+    "L\225\186\161c \196\145\195\160";
+    "\224\176\146\224\176\130\224\176\159\224\177\134";
+    "\224\174\146\224\174\159\224\175\141\224\174\159\224\174\149\224\
+     \174\174\224\175\141";
+    "\217\136\216\180\216\170\216\177";
+    "\224\164\137\224\164\183\224\165\141\224\164\159\224\165\141\224\
+     \164\176\224\164\131";
+    "\216\167\217\186";
```

Comment:
FTR, I got trouble with the text editor getting confused and changing characters under my feet, but for a test file go ahead and use UTF-8 in comments.